src/librawspeed/common/RawImage.h

```diff
@@ -112,24 +112,24 @@ class RawImageData
   uint32 getCpp() const { return cpp; }
   uint32 getBpp() const { return bpp; }
   void setCpp(uint32 val);
-  virtual void createData();
+  virtual void createData() final;
   virtual void destroyData();
   void blitFrom(const RawImage& src, const iPoint2D& srcPos,
                 const iPoint2D& size, const iPoint2D& destPos);
   RawSpeed::RawImageType getDataType() const { return dataType; }
   uchar8* getData();
   uchar8* getData(uint32 x, uint32 y);    // Not super fast, but safe. Don't use per pixel.
   uchar8* getDataUncropped(uint32 x, uint32 y);
-  virtual void subFrame(iRectangle2D cropped );
-  virtual void clearArea(iRectangle2D area, uchar8 value = 0);
+  virtual void subFrame(iRectangle2D cropped) final;
+  virtual void clearArea(iRectangle2D area, uchar8 value = 0) final;
   iPoint2D getUncroppedDim();
   iPoint2D getCropOffset();
   virtual void scaleBlackWhite() = 0;
   virtual void calculateBlackAreas() = 0;
   virtual void setWithLookUp(ushort16 value, uchar8* dst, uint32* random) = 0;
-  virtual void sixteenBitLookup();
-  virtual void transferBadPixelsToMap();
-  virtual void fixBadPixels();
+  virtual void sixteenBitLookup() final;
+  virtual void transferBadPixelsToMap() final;
+  virtual void fixBadPixels() final;
```

Comment:
Please remove the `virtual` keyword here (and in similar places) instead of adding `final`. This is a bog standard method, see e.g. `getCropOffset()` above. Declaring something `virtual` and `final` in the base class makes no sense to me. (There have been way too many virtual keywords spread around the code).